File: 02_Programming_Fundamentals/02_Programming_Fundamentals_with_C#/02_Intro_And_Basic_Syntax_Exercise/07_VendingMachine.cs
using System;

public class VendingMachine 
{

  public static void Main () 
  {

    string inputCoins = "";
    decimal coins = 0;
    decimal totalCoins = 0;

    string inputProducts = "";
    decimal nutsPrice = (decimal)2.0;
    decimal waterPrice = (decimal)0.7;
    decimal crispsPrice = (decimal)1.5; 
    decimal sodaPrice = (decimal)0.8;
    decimal cokePrice = (decimal)1.0;

    while (true) 
    {
      inputCoins = Console.ReadLine();
      if (inputCoins == "Start") { break; }
      coins = decimal.Parse(inputCoins);

      if (coins == (decimal)0.1 || coins == (decimal)0.2 || coins == (decimal)0.5 || coins == (decimal)1 || coins == (decimal)2) 
      {
        totalCoins += coins;
      } 
      else 
      {
        Console.WriteLine("Cannot accept {0}", coins);
      }
    }

    while (true) 
    {
      inputProducts = Console.ReadLine();
      if (inputProducts == "End") { break; }

      if (inputProducts == "Nuts") 
      {
        if (totalCoins >= nutsPrice) 
        { 
          Console.WriteLine("Purchased nuts"); 
          totalCoins -= nutsPrice;
        } 
        else 
        { 
          Console.WriteLine("Sorry, not enough money"); 
        }
      }
      else if (inputProducts == "Water") 
      {
        if (totalCoins >= waterPrice) 
        { 
          Console.WriteLine("Purchased water");
          totalCoins -= waterPrice; 
        } 
        else 
        { 
          Console.WriteLine("Sorry, not enough money"); 
        }
      }
      else if (inputProducts == "Crisps") 
      {
        if (totalCoins >= crispsPrice) 
        { 
          Console.WriteLine("Purchased crisps"); 
          totalCoins -= crispsPrice;
        } 
        else 
        { 
          Console.WriteLine("Sorry, not enough money"); 
        }
      }
      else if (inputProducts == "Soda") 
      {
        if (totalCoins >= sodaPrice) 
        { 
          Console.WriteLine("Purchased soda"); 
          totalCoins -= sodaPrice;
        } 
        else 
        { 
          Console.WriteLine("Sorry, not enough money");
        }
      }
      else if (inputProducts == "Coke") 
      {
        if (totalCoins >= cokePrice) 
        { 
          Console.WriteLine("Purchased coke"); 
          totalCoins -= cokePrice;
        } 
        else 
        { 
          Console.WriteLine("Sorry, not enough money"); 
        }
      }
      else 
      {
        Console.WriteLine("Invalid product");
      }
    }

    Console.WriteLine("Change: {0:F2}", totalCoins);

  }

}

File: 02_Programming_Fundamentals/04_Programming_Fundamentals_PHP/02_Intro_And_Basic_Syntax_Exercise/07_Vending_Machine.php
<?php 

$inputCoins = "";
$coins = 0;
$totalCoins = 0;

$inputProducts = "";
$nutsPrice = 2.0;
$waterPrice = 0.7;
$crispsPrice = 1.5; 
$sodaPrice = 0.8;
$cokePrice = 1.0;

while (true) 
{
  $inputCoins = readline();
  if ($inputCoins == "Start") { break; }
  $coins = floatval($inputCoins);

  if ($coins == 0.1 || $coins == 0.2 || $coins == 0.5 || $coins == 1 || $coins == 2) {
    $totalCoins += $coins;
  } else {
    echo "Cannot accept $coins" . PHP_EOL;
  }
}

while (true) 
{
  $inputProducts = readline();
  if ($inputProducts == "End") { break; }

  if ($inputProducts == "Nuts") {
    if ($totalCoins >= $nutsPrice) { echo "Purchased nuts" . PHP_EOL; $totalCoins -= $nutsPrice; } 
    else { echo "Sorry, not enough money" . PHP_EOL; }
  } else if ($inputProducts == "Water") {
    if ($totalCoins >= $waterPrice) 
    { 
      echo "Purchased water" . PHP_EOL;
      $totalCoins -= $waterPrice; 
    } 
    else 
    { 
      echo "Sorry, not enough money" . PHP_EOL; 
    }
  }
  else if ($inputProducts == "Crisps") 
  {
    if ($totalCoins >= $crispsPrice) 
    { 
      echo "Purchased crisps" . PHP_EOL; 
      $totalCoins -= $crispsPrice;
    } 
    else 
    { 
      echo "Sorry, not enough money" . PHP_EOL; 
    }
  }
  else if ($inputProducts == "Soda") 
  {
    if ($totalCoins > $sodaPrice) 
    { 
      echo "Purchased soda" . PHP_EOL; 
      $totalCoins -= $sodaPrice;
    } else if ((string)$totalCoins == (string)$sodaPrice) {
      echo "Purchased soda" . PHP_EOL; 
      $totalCoins -= $sodaPrice;
    }
    else 
    { 
      echo "Sorry, not enough money" . PHP_EOL;
    }
  }
  else if ($inputProducts == "Coke") 
  {
    if ($totalCoins >= $cokePrice) 
    { 
      echo "Purchased coke" . PHP_EOL; 
      $totalCoins -= $cokePrice;
    } 
    else 
    { 
      echo "Sorry, not enough money" . PHP_EOL; 
    }
  }
  else 
  {
    echo "Invalid product" . PHP_EOL;
  }
}

$totalCoins = number_format($totalCoins, 2, '.', '');
if ($totalCoins < 0.1) {
  $totalCoins = number_format(abs($totalCoins), 2, '.', '');
  echo "Change: $totalCoins" . PHP_EOL;
} else {
  echo "Change: $totalCoins" . PHP_EOL;
}

?>